text/0000-cosign.md

```diff
@@ -0,0 +1,109 @@
+# Meta
+[meta]: #meta
+- Name: Cosign support
+- Start Date: 2021-12-03
+- Author(s): [@samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Adds image signing support to Buildpacks natively using cosign.
+
+# Definitions
+[definitions]: #definitions
+
+- cosign: A sigstore project to sign container images - https://github.com/sigstore/cosign
+- SBOM: Software Bill-of-Materials
+
+
+# Motivation
+[motivation]: #motivation
+
+Supply chain security has been on top of everyone's minds in 2021. It is important to ensure the integrity and origin of software artifacts in order to have a secure supply chain. Sigstore's cosign has quikcly risen to become the defacto container signing solution. This RFC proposes that we natively add cosign support to the lifecycle to facilitate easy signed container artifacts as part of the build process.
+
+# What it is
+[what-it-is]: #what-it-is
+
+As a part of the exporting process, the `analyzer` and `exporter` will now take an addition flag `-cosign-config <cosign.toml>` which will contain the cosign configuration.
+
+The schema for `cosign.toml` will look like the following - 
+
+```toml
+# A static list of annotations to add to each signature config
+[[annotations]]
+name = "<name>"
+value = "<value>"
+
+[[annotations]]
+name = "<name2>"
+value = "<value2>"
+
+# multiple cosign configs can be specified
+[[configs]]
+# path to private key
+key = "<path-to-key>"
+# Exclusive OR with `key`
+# private key inline as a b64 encoded string
+key-data = "<base64-encoded-private-key>"
+# optional value for COSIGN_REPOSITORY, optional, default:""
+repository = "<cosign-repository>"
+# value for COSIGN_DOCKER_MEDIA_TYPES, optional, default:false
+docker-media-types = true
+# optional path to password file. Contents will be used as COSIGN_PASSWORD
+password = "<path-to-password-file>"
+# Exclusive OR with `password`
+# optional base64 encoded value for COSIGN_PASSWORD
+password-data = "<base64-encoded-password>"
```

Comment:
I really rather not put this here at all. I think we should require the `COSIGN_PASSWORD` to be provided by the platform.

Reply:
Is there a similar concern with the base64 encoded private key?

Reply:
I have the same broad concern about _requiring_ (as an additional option to one that does not require that data to rest, is fine) any sensitive data to be encoded into files.  Can we please explore some additional styles of contributing all of this sensitive information without requiring it to be in a file owned by us, on the file system?

Reply:
Yes, I agree with @jabrown85 and @nebhale here, those should be injected by the platform.